pkg/ipam/clusterpool.go

```diff
@@ -154,12 +155,31 @@ func newCRDWatcher(conf Configuration, nodeWatcher nodeWatcher, owner Owner, nod
 	}
 
 	// Subscribe to CiliumNode updates
-	nodeWatcher.RegisterCiliumNodeSubscriber(c)
+	c.eventloop(nodes)
 	owner.UpdateCiliumNodeResource()
 
 	return c
 }
 
+func (c *crdWatcher) eventloop(res agentK8s.LocalCiliumNodeResource) {
+	// Ensure that the subscrption happens synchronously (but processing
+	// asynchronously), so that there's no window in which we miss events.
+	// FIXME is that actually a race?
+	// Don't retry events - the downstream code isn't setup to handle retries.
+	evs := res.Events(context.TODO(), resource.WithErrorHandler(resource.RetryUpTo(0)))
```

Comment:
Same here as in previous comments: use a proper context that is cancelled when this sub-system is stopping. Or if there is no such thing, validate that it's fine that this only stops when the `Resource[T]` stops and it's fine handling events from it while everything is being stopped & add comment to explain this.

Reply:
clusterpool was removed